railgun/manager/manager.go

```diff
@@ -0,0 +1,205 @@
+// Package manager implements the controller manager for all controllers in Railgun.
+package manager
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"net/http"
+	"os"
+
+	"github.com/kong/go-kong/kong"
+	"github.com/kong/kubernetes-ingress-controller/pkg/sendconfig"
+	konghqcomv1 "github.com/kong/kubernetes-ingress-controller/railgun/apis/configuration/v1"
+	"github.com/kong/kubernetes-ingress-controller/railgun/apis/configuration/v1alpha1"
+	configurationv1alpha1 "github.com/kong/kubernetes-ingress-controller/railgun/apis/configuration/v1alpha1"
+	configurationv1beta1 "github.com/kong/kubernetes-ingress-controller/railgun/apis/configuration/v1beta1"
+	"github.com/kong/kubernetes-ingress-controller/railgun/controllers"
+	kongctrl "github.com/kong/kubernetes-ingress-controller/railgun/controllers/configuration"
+	"k8s.io/apimachinery/pkg/runtime"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/tools/clientcmd"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/healthz"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+)
+
+// Config collects all configuration that the controller manager takes from the environment.
+// BUG: the above is not 100% accurate today - controllers read some settings from environment variables directly
+type Config struct {
+	MetricsAddr          string
+	EnableLeaderElection bool
+	ProbeAddr            string
+	KongURL              string
+	FilterTag            string
+	Concurrency          int
+	SecretName           string
+	SecretNamespace      string
+	KubeconfigPath       string
+
+	ZapOptions zap.Options
```

Comment:
lint: while I understand we're mostly moving things around, technically public symbols for this type should have `godoc` comments

Reply:
While I agree that all exported types should have `godoc` comments, duplication of documentation between docstrings and flag descriptions would be redundant.

Did https://github.com/Kong/kubernetes-ingress-controller/pull/1175/files#r613244057 as a compromise for now.